server/app/services/geo/esri/EsriClient.java

```diff
@@ -262,36 +269,91 @@ CompletionStage<Optional<JsonNode>> fetchServiceAreaFeatures(
 
   /**
    * Calls an external Esri service to get the service areas of the provided {@link
-   * AddressLocation}. Takes the returned service areas and returns a boolean specifying if the
-   * service area is included.
-   *
-   * @return an optional boolean if successful, or an empty optional if the request fails.
+   * AddressLocation}. Takes the returned service areas and returns an immutable list of {@link
+   * EsriServiceAreaInclusion}, filtered by the services areas specified in the application config.
    */
-  public CompletionStage<Optional<Boolean>> isAddressLocationInServiceArea(
+  public CompletionStage<ImmutableList<EsriServiceAreaInclusion>> getServiceAreaInclusionGroup(
       EsriServiceAreaValidationOption esriServiceAreaValidationOption, AddressLocation location) {
     return fetchServiceAreaFeatures(location, esriServiceAreaValidationOption.getUrl())
         .thenApply(
             (maybeJson) -> {
+              EsriServiceAreaInclusion.Builder esriServiceAreaInclusionBuilder =
+                  EsriServiceAreaInclusion.builder();
+              ImmutableList.Builder<EsriServiceAreaInclusion> inclusionListBuilder =
+                  ImmutableList.builder();
               if (maybeJson.isEmpty()) {
                 logger.error(
                     "EsriClient.fetchServiceAreaFeatures JSON response is empty. Called by"
                         + " EsriClient.isAddressLocationInServiceArea."
                         + " EsriServiceAreaValidationOption = {}. AddressLocation = {}",
                     esriServiceAreaValidationOption,
                     location);
-                return Optional.empty();
+                inclusionListBuilder.add(
+                    esriServiceAreaInclusionBuilder
+                        .setArea(esriServiceAreaValidationOption.getId())
+                        .setState(EsriServiceAreaState.FAILED)
+                        .setTimeStamp(Instant.now())
+                        .build());
```

Comment:
**question:** should we store a failure entry for each option in the config?

Reply:
That makes sense to me.

Reply:
Looks like it's still storing just the one